### DISCUSSION TITLE
How to use pyautofep calculation when the ligand is peptide

### DISCUSSION THREAD

**Discussion by CPU-HORDB** [luancarvalhomartins/PyAutoFEP#103] [General]

**Comment by luancarvalhomartins** [comment]
Dear @CPU-HORDB,

Thank you for you message. I am moving it to the discussions, as I think it is more suitable to that.

**Comment by luancarvalhomartins** [comment]
Originally, PyAutoFEP was not written for peptides, although I agree it would be quite useful.

For using it as is, the complexity will depend on what parameters you want to use.

- The fast and dirty way would parameterizing the peptide as a small molecule the using the regular workflow. It should work, but the parameters will probably be worse than the FF parameters.
- The correct way would be to obtain parameters for the peptide using the FF you will use and than pass that topology as ligand topology. You may experiment with `gmx pdb2gmx` and `gmx grompp -pp`, then trimming the resulting topology to keep only the peptide parameters.

Note that obtaining an MCS will be challenging. Maybe you want to pass `mcs_custom_mcs`. In case you are inverting stereocenters, it will be extra challenging and I would guess `find_mcs_3d` will fail. In case there are no inversions, maybe `find_mcs` will work. Superposition may work, it is hard to know before hand. The larger the peptide, the harder the superposition. In case superposition fails, LMK. I could try coding the option to use input poses as pose.

Feel free to keep in touch regarding this.

**Comment by ninomiya829** [comment]
Dear @luancarvalhomartins,

I've recently embarked on using PyAutoFEP to estimate the FEP for a cyclic peptide comprising roughly eight amino acids, along with its derivatives with related proteins. 

However, after executing the command, ` generate_perturbation_map.py --map_type=star --map_bias=1918_lig --input lig_data/1918_lig.mol lig_data/1919_lig.mol lig_data/1920_lig.mol` 

I've noticed that the program seems to be stuck at a certain point for an unusually long time. 

It seems stucking in `mcs_data = os_util.starmap_unpack(find_mcs, thread_pool,[[[molecules[mol_i], molecules[mol_j]], None, verbosity]for (mol_i, mol_j) in todo_pairs],itertools.repeat({'completeRingsOnly': True, 'matchValences': True,'ringMatchesRingOnly': True}))` step

It's got me wondering if PyAutoFEP might not be the ideal tool for handling such a large cyclic peptide, or perhaps I've overlooked something in my setup.

Would you happen to know if there's a workaround or a particular adjustment I could make to tackle this issue? I'd really appreciate your insights and assistance on this. Thanks a bunch for taking the time to help me out!
